src/Clapi/Relay.hs

```diff
@@ -92,6 +93,25 @@ genInitDigest ps tns vs =
         RtnDataSeries ts ->
           d'{ocdData = alInsert p (oppifyTimeSeries ts) $ ocdData d}
 
+contDiff
+  :: AssocList Seg (Maybe Attributee) -> AssocList Seg (Maybe Attributee)
+  -> Map Seg (Maybe Attributee, SequenceOp Seg)
+contDiff a b = afters <> absents
+  where
+    asAfter (acc, prev) k ma = ((k, (ma, SoPresentAfter prev)) : acc, Just k)
+    aftersFor = Map.fromList . fst . alFoldlWithKey asAfter ([], Nothing)
+    aftersA = aftersFor a
+    aftersB = aftersFor b
+    afters = Map.difference aftersB aftersA
+    absents = fmap (const SoAbsent) <$> Map.difference aftersA aftersB
+
+mayContDiff :: Maybe (RoseTreeNode a) -> AssocList Seg (Maybe Attributee) -> Maybe (Map Seg (Maybe Attributee, SequenceOp Seg))
+mayContDiff ma kb = case ma of
+    Just (RtnChildren ka) -> if ka == kb
+        then Nothing
+        else Just $ contDiff ka kb
```

Comment:
Is it the case that the equality check above is roughly as costly as doing the diff anyway? In which case this could reformulated to checking whether the result of the diff is null to return `Nothing`?

Reply:
It could indeed, thought about it and concluded that since couldn't say I know what the performance tradeoff was I was going to go with the simplest thing.